src/Pos/Communication/Message.hs

```diff
@@ -49,258 +49,303 @@ instance Message MsgSubscribe where
 
 instance Message k => Message (ReqMsg (Tagged k v)) where
     messageCode _ = messageCode (Proxy :: Proxy k)
-    formatMessage _ = "Tagged"
+    formatMessage _ = "Tagged ReqMsg"
+
+instance Message k => Message (ReqOrRes (Tagged k v)) where
+    messageCode _ = messageCode (Proxy :: Proxy k)
+    formatMessage _ = "Tagged ReqOrRes"
 
 instance Message k => Message (MempoolMsg (Tagged k v)) where
     messageCode _ = messageCode (Proxy :: Proxy k)
-    formatMessage _ = "Tagged"
+    formatMessage _ = "Tagged MempoolMsg"
 
 instance Message k => Message (DataMsg (Tagged k v)) where
     messageCode _ = messageCode (Proxy :: Proxy k)
-    formatMessage _ = "Tagged"
+    formatMessage _ = "Tagged DataMsg"
 
 instance Message k => Message (InvMsg (Tagged k v)) where
     messageCode _ = messageCode (Proxy :: Proxy k)
-    formatMessage _ = "Tagged"
+    formatMessage _ = "Tagged InvMsg"
 
 instance Message k => Message (InvOrData key (Tagged k v)) where
     messageCode _ = messageCode (Proxy :: Proxy k)
-    formatMessage _ = "Tagged"
+    formatMessage _ = "Tagged InvOrData"
 
 
 instance Message (ReqMsg TxMsgContents) where
     messageCode _ = 32
     formatMessage _ = "Request"
 
-instance Message (MempoolMsg TxMsgContents) where
+instance Message (ReqOrRes TxMsgContents) where
     messageCode _ = 33
+    formatMessage _ = "ReqOrRes"
+
+instance Message (MempoolMsg TxMsgContents) where
+    messageCode _ = 34
```

Comment:
#1288 was recently merged. `Word16` identifies a conversation now, rather than `ByteString` as it was before. Prior to that, most of these instances were derived from a `MessagePart` class, which also gave a `ByteString`: the `MessageName` of the `ReqMsg` or related type was `<>`ed with the `MessageName` of the `MessagePart` à la

`instance MessagePart thing => Message (ReqMsg thing) where ...`

But we don't do that anymore. We *could* with some high bits / low bits scheme, but I think it's better to list the codes explicitly. We shouldn't need codes for every one of these things. Codes ought to identify *conversations* instead.